### PR TITLE
Jakarta Servlet 6.1 Release

### PR DESCRIPTION
**Pull request by markt-asf**
## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [X] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [X] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   - ~[ ] Generated IP Log~ *no longer required according to project handbook*
   - [x] Email to PMC
   - [x] Start release review by emailing EMO 
- [X] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/jakartaservlet-1026/
- [X] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/servlet/jakartaee11/staged/eftl/jakarta-servlet-tck-6.1.0.zip
- [X] The URL of the compatibility certification request issue:
      https://github.com/jakartaee/servlet/issues/642
- [X] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

**Comment by markt-asf** [comment]
The TCK is currently only available in a staged Maven repo. This review should not progress until a 'standard' TCK zip is staged to 
https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee11/staged/

**Comment by arjantijms** [comment]
> The TCK is currently only available in a staged Maven repo. This review should not progress until a 'standard' TCK zip is staged to https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee11/staged/

Hi, I took a look at the current TCK. At the last platform call we thought it was just a matter of copying a zip file from Maven to the download location, but there's a little bit more needed.

There is not really a "standard" TCK zip, unfortunately, but maybe REST and Pages are good examples. They both contain folder as follows:

```
artefacts
    jakarta-foo-tck-version.jar
    jakarta-foo-tck-version-sources.jar
docs
    html-usersguide
    pdf-usersguide
    assertions
    [...]
LICENSE.md
```

In the Servlet TCK folder we don't have any TCK user guide, and two folders with a tck-runtime.jar and a tck-utils.jar. I guess an assembly file is needed to create a zip file with those two artefacts, and then we have to add a basic user guide still.

**Comment by edburns** [comment]
Please see https://github.com/jakartaee/servlet/issues/617.

**Comment by markt-asf** [comment]
Updates to the TCK are complete. Waiting on creation of the download directory so CI can generated the staged package.

**Comment by markt-asf** [comment]
TCK has been re-staged and this PR has been updated. As far as I am aware, this is now ready for ballot.

**Comment by marceloancelmo** [comment]
> TCK has been re-staged and this PR has been updated. As far as I am aware, this is now ready for ballot.

Thanks @markt-asf. I am checking it now and will start the ballot

Best Regards, Marcelo Ancelmo

**Comment by ivargrimstad** [comment]
@marceloancelmo Please complete the Mentor Spec Review Checklist in the comment above as a help for you when reviewing the PR and visualizing progress.

**Comment by ivargrimstad** [comment]
@markt-asf  Can you fix the links in the _index.md file so I can get the ballot underway?
Or if you can tick the "maintainers can change" on the PR so I can fix it for you?

**Comment by ivargrimstad** [comment]
> "Allow edits by maintainers" is ticked. It isn't immediately obvious to me what needs to change else I'd fix it.

I'll check it out. and make the changes. It is a little confusing that the links need to be pointing to the location the files will be prompted to, and not the location they are at right now....

**Comment by ivargrimstad** [comment]
Done!
Servlet is now ready for ballot. Thanks @markt-asf!

**Comment by kazumura** [comment]
If there is a chance to fix, please chnage the copyright year to "2019,2024" in the javadocs footer.

**Comment by markt-asf** [comment]
I'll update the Javadoc by the end of the day.

**Comment by markt-asf** [comment]
Given it will need a new release of the API to fix the Javadoc, I'm going to do a new TCK release as well (and CCR) to pick up the fixes for a couple issues found today. Should still all be in place by end of day.

File: servlet/6.1/_index.md [hunk 1]
```diff
@@ -19,31 +19,28 @@ The release removes references to the SecurityManager and provides various small
...
 
 * [Jakarta Servlet 6.1 Release Record](https://projects.eclipse.org/projects/ee4j.servlet/releases/6.1)
-
-<!--
-The following can be uncommented and version information updated as they become available.
-
-  * [Jakarta EE Platform 10 Release Plan](https://jakartaee.github.io/platform/jakartaee10/JakartaEE10ReleasePlan)
-* [Jakarta Servlet 6.0 Specification Document](./jakarta-servlet-spec-6.0.pdf) (PDF)
-* [Jakarta Servlet 6.0 Specification Document](./jakarta-servlet-spec-6.0.html) (HTML)
+  * [Jakarta EE Platform 11 Release Plan](https://jakartaee.github.io/platform/jakartaee11/JakartaEE11ReleasePlan)
```
Comment (Pandrex247) [inline servlet/6.1/_index.md:27]: Missing plan link for Servlet
```suggestion
* [Jakarta Servlet 6.1 Release Plan](https://projects.eclipse.org/projects/ee4j.servlet/releases/6.1.0/plan)
  * [Jakarta EE Platform 11 Release Plan](https://jakartaee.github.io/platform/jakartaee11/JakartaEE11ReleasePlan)
```

**Review by Pandrex247 — COMMENTED** [review]
A couple of corrections

**Comment by pnicolucci** [comment]
@markt-asf it would be great to update the new features to include:

1) Add the jakarta.servlet.request.secure_protocol attribute to the list of request attributes that servlet containers are required to populate if the request has been transmitted over a secure protocol.
2) Require that all error dispatches are performed as HTTP GET requests with the original HTTP method being made available via a new request attribute jakarta.servlet.error.method.
3) Provide a mechanism to allow applications to interact with the HttpSession object outside of the standard HTTP request processing. This is expected to be particularly useful to applications using the Jakarta WebSocket API.

From: https://github.com/jakartaee/servlet/blob/master/spec/src/main/asciidoc/servlet-spec-body.adoc#appendix-a-change-log

**Comment by ivargrimstad** [comment]
On ballot completion, the specification committee mentor:

 - [ ] adds this final checklist to the main PR.
 - [ ] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
https://github.com/jakartaee/servlet/issues/651

**Comment by mtdelgadoa** [comment]
## EMO REVIEW CHECKLIST 

EDP Review status: COMPLETED 

### **EMO review checklist**
PMI record: https://projects.eclipse.org/projects/ee4j.servlet/reviews/6.1.0-release-review

_**EF Specification Process**_      
- [x] Spec Committee Ballot completed

_**Intellectual Property Management**_      
- [ ] All project code has [copyright and license headers](https://www.eclipse.org/projects/handbook/#ip-copyright-headers) correctly applied. ** EMO will scan the code at their discretion **   
- [x]  All distributed third-party content has been vetted by the IP Due Diligence process (i.e., IP Log has been approved)   

_**Open Source Rules of Engagement**_
- [x] PMC approval [on the mailing list](https://github.com/eclipsefdn/emo-team/emo/-/blob/main/%7Blink%20to%20the%20message%7D)

General:
- [x] Project is operating within the mission and scope defined in its top-level project’s charter   
- [x]  Project is operating within the bounds of its own scope   
- [x]  Project is operating in an open and transparent manner       
- [x]  Overall the project is operating according to the Eclipse Development Process.   

Things to check:
- Communication channels advertised
- Advertised communication channels used
- Committers are responding to questions
- Committers are responding to issues
- Committers are responding to pull/merge/review requests

_**Branding and Trademarks**_
The following applies when the project has a custom website.
To the best of our knowledge:
- [x]  Project content correctly uses Eclipse Foundation trademarks    
- [x] Project content (code and documentation) does not violate trademarks owned by other organizations   

Things to check:
- [ ]  Project website uses the [project's formal name](https://www.eclipse.org/projects/handbook/#trademarks-website-name) in first and all prominent references
- [ ]   Project website includes a trademark attribution statement
- [ ] Project website footers contain all [necessary elements](https://www.eclipse.org/projects/handbook/#trademarks-website-footer)

_**Legal Documentation**_
Required files:   
- [x] License files in all repository roots   
- [x]  README       
- [x]  CONTRIBUTING (or equivalent)   

Recommended files:
- [x] NOTICES or equivalent (you can use the [Legal Documentation generator](https://www.eclipse.org/projects/tools/documentation.php) available under committer tools)   
- [x]  CODE_OF_CONDUCT   
- [ ]  SECURITY   
 
See examples for [Security file](https://gitlab.eclipse.org/security/best-practices/-/blob/main/templates/SECURITY.md) and [Code of Conduct](https://github.com/eclipse/.github/blob/master/CODE_OF_CONDUCT.md).

Required elements:
- [x]  ECA is referenced/described   

Recommended elements:
- [x]  Build instructions provided   
- [ ] [Security policy](https://www.eclipse.org/security/policy.php) is described   
 
_**Metadata (PMI)**_
- [x] The formal name, e.g. "Eclipse Foo™", is used in the project title   
- [x] The formal name including appropriate marks (e.g, "™") is used in the first mention in the text of the project description, and scope   
- [x] The project description starts with a single paragraph that can serve as an executive summary   
- [x] Source code repository references are up-to-date   
- [ ] Download links and information are up-to-date (see [EF handbook for more information on how-to do this](https://www.eclipse.org/projects/handbook/#resources-downloads))   
- [x] Communication channels listed in the PMI (i.e. public mailing list, forums, etc.)

**Pull request merged**